### PR TITLE
PhpCheatSheet: Change order of functions according to relevancy

### LINKED ISSUE #3183: PHP Cheat Sheet: Need improvement
There is lot of functions in this cheat sheet which nobody uses in day to day life. I guess cheat sheet are meant to be useful for quick summarizing the concept. There is no MySQL functions which is used in daily basis. And why we need Regular expression in this cheat sheet.

Need improvement

------
IA Page:  http://duck.co/ia/view/php_cheat_sheet 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @rathoreanirudh

### PR DESCRIPTION
**Pull request by sahildua2305**
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.
- [x] Improvement
    - [x] Bug fix: **`PhpCheatSheet: Fix 3183`**

###### Description of changes
I feel that many functions under sections PHP array functions and string functions weren't in proper order. I have sorted the functions according to their relevancy.

###### Which issues (if any) does this fix?

Fixes #3183 

###### People to notify (@mention interested parties)

---

Instant Answer Page: https://duck.co/ia/view/php_cheat_sheet 

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @rathoreanirudh

### CODE REVIEW

**Comment by daxtheduck** [comment]
@sahildua2305 Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by sahildua2305** [comment]
@alohaas @shahbaz17 Please review the quick changes.

File: share/goodie/cheat_sheets/json/php.json [hunk 1]
```diff
@@ -62,12 +62,15 @@
...
         }],
         "PHP String Functions":[{
-            "val": "return part of a string",
-            "key": "substr()"
-        }, {
             "val": "output one or more string",
             "key": "echo"
         }, {
+            "val": "output a string",
+            "key": "print"
```
Comment (gaulrobe) [inline share/goodie/cheat_sheets/json/php.json]: `print` isn't _really_ a function and can be called with or without `()`... should this live in a different section?

Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/php.json]: @gaulrobe Technically it's still a function.
Reference - http://php.net/manual/en/function.print.php

Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/php.json]: And I don't think it should go in a different section. @gaulrobe 
What section will you suggest, if that was the case?

Comment (gaulrobe) [inline share/goodie/cheat_sheets/json/php.json]: From that link:
`print is not actually a real function (it is a language construct) so you are not required to use parentheses with its argument list.`

Maybe there should be a `built-ins` or `internal construct` section?

Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/php.json]: Point! This way even `echo` isn't a function. It's also a language construct.

Comment (gaulrobe) [inline share/goodie/cheat_sheets/json/php.json]: Yep, exactly! Let's make a new section for these? 😄

Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/php.json]: Cool! Making one. :smiley:

Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/php.json]: @gaulrobe Can you please point me at some exhaustive reference which can be used to list maximum (mostly used) language constructs? I can't recall more than 3-4.

Comment (gaulrobe) [inline share/goodie/cheat_sheets/json/php.json]: @sahildua2305 there isn't a really good list of built-ins (or I'm just not good at finding one 😉)... but having `print` and `echo` and any other ones you are thinking in here is a good start!

Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/php.json]: Already added almost 7 of the most commonly used ones. Will push after testing.

**Comment by sahildua2305** [comment]
@gaulrobe Check now. :wink:

File: share/goodie/cheat_sheets/json/php.json [hunk 2]
```diff
@@ -1,21 +1,53 @@
...
+            "val": "Determine whether a variable is empty",
+            "key": "empty"
+        }, {
+            "val": "Include and evaluate the specified file",
+            "key": "include"
+        }, {
+            "val": "Include and evaluate the specified file, raise CompileError upon failure",
+            "key": "require"
+        }, {
+            "val": "Equivalent to exit",
```
Comment (gaulrobe) [inline share/goodie/cheat_sheets/json/php.json]: We should add `exit` to this cheat sheet if we are going to reference it here.

**Comment by gaulrobe** [comment]
@sahildua2305 this is looking pretty good to me.

File: share/goodie/cheat_sheets/json/php.json (hunk 2 above)
Comment (sahildua2305) [inline share/goodie/cheat_sheets/json/php.json]: Point! Updated.

**Comment by sahildua2305** [comment]
@moollaza Please review.

**Comment by sahildua2305** [comment]
@pjhampton have a look at this please.

**Comment by edgesince84** [comment]
thanks Sahil! --  also, any input here @rathoreanirudh?

**Comment by sahildua2305** [comment]
@edgesince84 @pjhampton I think the maintainer isn't active these days and hence isn't responding. Is there no way to proceed if he doesn't respond?

**Comment by pjhampton** [comment]
Hey @sahildua2305, 

Sure, sometimes life / work takes us away from our machines and maybe @rathoreanirudh just isn't available at this time. I believe it's 14 days since first contact (7 days for high priority) See [the docs](http://docs.duckduckhack.com/maintaining/guidelines.html#for-maintainers).

My advice is be patient :) We will get there!

**Comment by sahildua2305** [comment]
Ok. Thanks @pjhampton :smile:

**Comment by sahildua2305** [comment]
@moollaza Let's get ahead with this? I'd like to take over as a maintainer if no one has any issues :smile:

**Comment by sahildua2305** [comment]
@moollaza Is there any issue with this one?

**Comment by moollaza** [comment]
@sahildua2305 Thanks! This LGTM :+1:

**Comment by sahildua2305** [comment]
@moollaza Can I take over as maintainer for this IA? PHP has been my major language for last 3 years. I think I can do a good job with this one :)

**Comment by edgesince84** [comment]
@rathoreanirudh do you want to relinquish maintainership for this IA?

cc @sahildua2305

**Pull request merged**